app/services/change_lgsl_code_service.rb

```diff
@@ -0,0 +1,43 @@
+class ChangeLgslCodeService
+  class InvalidLGSLCodeError < StandardError
+    def message
+      "An LGSL code must be present and an integer"
+    end
+  end
+
+  class LocalServiceDoesNotExistError < StandardError
+    def message
+      "The LocalService you are trying to change the LGSL code of does not exist"
+    end
+  end
+
+  class LocalServiceAlreadyExistsError < StandardError
+    def message
+      "The LocalService you are trying to change the LGSL code to already exists"
+    end
+  end
+
+  def self.call(from_lgsl: nil, to_lgsl: nil)
+    raise InvalidLGSLCodeError unless from_lgsl.present? && from_lgsl.is_a?(Integer)
+
+    raise InvalidLGSLCodeError unless to_lgsl.present? && to_lgsl.is_a?(Integer)
+
+    raise LocalServiceDoesNotExistError if LocalService.where(lgsl_code: from_lgsl).empty?
+
+    raise LocalServiceAlreadyExistsError if LocalService.where(lgsl_code: to_lgsl).any?
```

Comment:
This is impressively thorough but I think it's a bit too much for a temporary task, I think you can avoid the need for much of these checks through Mongoid's API and validations.